src/github.com/getlantern/flashlight/client/handler.go

```diff
@@ -21,7 +24,16 @@ const (
 // handler available from getHandler() and latest ReverseProxy available from
 // getReverseProxy().
 func (client *Client) ServeHTTP(resp http.ResponseWriter, req *http.Request) {
-	logging.RegisterUserAgent(req.Header.Get("User-Agent"))
+	userAgent := req.Header.Get("User-Agent")
+	logging.RegisterUserAgent(userAgent)
+
+	ctx.SetAll(ctx.Map{
+		"op":         "proxy",
+		"user_agent": userAgent,
+		"request_id": rand.Int63(),
+		"origin":     req.Host,
+	})
+	defer ctx.Clear()
```

Comment:
IMHO, this is the major downside to this approach.  We have to remember to clear context state, otherwise we will leak memory.